sim/priest/talents.go

```diff
@@ -60,9 +60,591 @@ func (priest *Priest) ApplyTalents() {
 	// }
 
-	// if priest.Talents.TwistedFaith > 0 {
-	// 	priest.AddStatDependency(stats.Spirit, stats.SpellPower, 0.04*float64(priest.Talents.TwistedFaith))
-	// }
+	// Disciplin Talents
+	// Improved Power Word: Shield - TBD
+	// Twin Disciplines
+	if priest.Talents.TwinDisciplines > 0 {
+		priest.AddStaticMod(core.SpellModConfig{
+			School:     core.SpellSchoolHoly | core.SpellSchoolShadow,
+			ClassMask:  int64(PriestSpellsAll),
+			FloatValue: (0.02 * float64(priest.Talents.TwinDisciplines)),
+			Kind:       core.SpellMod_DamageDone_Pct,
+		})
+	}
+
+	// Mental Agillity
+	if priest.Talents.MentalAgility > 0 {
+		priest.AddStaticMod(core.SpellModConfig{
+			ClassMask:  int64(PriestSpellsAll),
+			FloatValue: -0.04 * float64(priest.Talents.MentalAgility),
+			Kind:       core.SpellMod_PowerCost_Pct,
+		})
+	}
+
+	// Evangelism
+	priest.applyEvangelism()
+
+	// Archangel
+	priest.applyArchangel()
+
+	// Shadow Talents
+	// Darkness
+	if priest.Talents.Darkness > 0 {
+		priest.PseudoStats.CastSpeedMultiplier *= 1 + (0.01 * float64(priest.Talents.Darkness))
+	}
+
+	// Improved Shadow Word: Pain
+	if priest.Talents.ImprovedShadowWordPain > 0 {
+		priest.AddStaticMod(core.SpellModConfig{
+			ClassMask:  int64(PriestSpellShadowWordPain),
+			FloatValue: 0.03 * float64(priest.Talents.ImprovedShadowWordPain),
+			Kind:       core.SpellMod_DamageDone_Flat,
+		})
+	}
+
+	// Veiled Shadows
+	if priest.Talents.VeiledShadows > 0 {
+		priest.AddStaticMod(core.SpellModConfig{
+			ClassMask: int64(PriestSpellFade),
+			TimeValue: time.Second * -3 * time.Duration(priest.Talents.VeiledShadows),
+			Kind:      core.SpellMod_Cooldown_Flat,
+		})
+
+		priest.AddStaticMod(core.SpellModConfig{
+			ClassMask: int64(PriestSpellShadowFiend),
+			TimeValue: time.Second * -30 * time.Duration(priest.Talents.VeiledShadows),
+			Kind:      core.SpellMod_Cooldown_Flat,
+		})
+	}
+
+	// Improved Psychic Scream
+	if priest.Talents.ImprovedPsychicScream > 0 {
+		priest.AddStaticMod(core.SpellModConfig{
+			ClassMask: int64(PriestSpellPsychicScream),
+			TimeValue: time.Second * -2 * time.Duration(priest.Talents.ImprovedPsychicScream),
+			Kind:      core.SpellMod_Cooldown_Flat,
+		})
+	}
+
+	// Improved Mind Blast
+	priest.applyImprovedMindBlast()
+
+	// Improved Devouring Plague
+	priest.applyImprovedDevouringPlague()
+
+	// Twisted Faith
+	if priest.Talents.TwistedFaith > 0 {
+		priest.AddStaticMod(core.SpellModConfig{
+			School:     core.SpellSchoolShadow,
+			ClassMask:  int64(PriestShadowSpells),
+			FloatValue: 0.01 * float64(priest.Talents.TwistedFaith),
+			Kind:       core.SpellMod_DamageDone_Pct,
+		})
+
+		priest.AddStatDependency(stats.Spirit, stats.SpellHit, 0.5*float64(priest.Talents.TwistedFaith))
+	}
+
+	// Shadowform
+	if priest.Talents.Shadowform {
+		// no class restrictions
+		priest.AddStaticMod(core.SpellModConfig{
+			School:     core.SpellSchoolShadow,
+			FloatValue: 0.15,
+			Kind:       core.SpellMod_DamageDone_Pct,
+		})
+	}
```

Comment:
Looking at the DB Shadowform comes up as a generic `Apply Aura: Mod Damage Done % (Shadow)` which should apply to ALL source of shadow damage and not just spells. So this should be `priest.PseudoStats.SchoolDamageDealtMultiplier[stats.SchoolIndexShadow] *= 1.15`

Reply:
I'm still getting used to the spell mods but in reality what you've done will do the same but instead of applying it to the generic unit pseudo stat it would apply to every single shadow school spell the priest registers (even non class ones). I guess I just need to get used to this way of defining mods 😅